src/herder/Upgrades.cpp

```diff
@@ -1029,4 +1242,201 @@ Upgrades::applyReserveUpgrade(AbstractLedgerTxn& ltx, uint32_t newReserve)
         prepareLiabilities(ltx, header);
     }
 }
+
+#ifdef ENABLE_NEXT_PROTOCOL_VERSION_UNSAFE_FOR_PRODUCTION
+ConfigUpgradeSetFrameConstPtr
+ConfigUpgradeSetFrame::makeFromKey(AbstractLedgerTxn& ltx,
+                                   ConfigUpgradeSetKey const& key)
+{
+    auto ltxe = ltx.loadWithoutRecord(ConfigUpgradeSetFrame::getLedgerKey(key));
+    if (!ltxe)
+    {
+        return nullptr;
```

Comment:
Just wanted to make sure, how the downstream code handles nullptr here? Would it just act as if we didn't have such upgrade armed?

Reply:
It depends. In the `CommandHandler`, we actually check after setting to make sure something was set. In `applyTo`, we throw an exception (this shouldn't happen, but if it does, it'll be caught). In `isValidForApply`, we return `XDR_INVALID`. In `isValidForNomination`, we return false.